crates/codegen/ebnf/src/ebnf_writer.rs

```diff
@@ -0,0 +1,76 @@
+#[derive(Debug)]
+#[allow(non_camel_case_types)]
+pub enum TokenKind {
+    comment,
+    constant,
+    keyword,
+    operator,
+    string,
+}
+
+pub trait EBNFWriter {
+    fn write_token(&mut self, kind: TokenKind, value: &str);
+    fn write_line_break(&mut self);
+
+    fn write_production_definition(&mut self, production_name: &str);
+    fn write_production_reference(&mut self, production_name: &str);
+    fn write_local_production_reference(&mut self, _parent_name: &str, production_name: &str) {
```

Comment:
Love `write_local_production_reference()`!
Was planning on doing it for the spec for named sub-expressions. Can reuse the same mechanism then!

Reply:
#332